packages/xsnap/src/xsnap.c

```diff
@@ -81,14 +83,13 @@ static char* fxWriteNetStringError(int code);
 // The order of the callbacks materially affects how they are introduced to
 // code that runs from a snapshot, so must be consistent in the face of
 // upgrade.
-#define mxSnapshotCallbackCount 6
+#define mxSnapshotCallbackCount 5
 txCallback gxSnapshotCallbacks[mxSnapshotCallbackCount] = {
 	fx_issueCommand, // 0
-	fx_Array_prototype_meter, // 1
-	fx_print, // 2
-	fx_setImmediate, // 3
-	fx_gc, // 4
-	fx_performance_now, // 5
+	fx_print, // 1
+	fx_setImmediate, // 2
+	fx_gc, // 3
+	fx_performance_now, // 4
 	// fx_evalScript,
```

Comment:
`NULL` was my hand-wave at a tombstone. Both you and @warner called my bluff, in different directions (which is good!).

The moddable folks added `fx_Function_prototype_bound` to the middle rather than the end of the array in `xsSnapshot.c`; so unless we want to push back on that, we're already not in a good position to preserve snapshot compatibility across this change.

> Are there any snapshots in databases that we care about continuing to be able to restore?

I think not, and @warner 's comment suggests likewise.

Reply:
agreed, not yet

We'll need some sort of versioning marker sooner or later. I suspect we're going to need `@agoric/xsnap` to keep multiple binaries around, since all of (snapshot binary format, the `callbacks` table, metering behavior) will be highly dependent upon the binary. So once a vat starts executing with a given version of the `xsnap` binary, both its snapshots and it's future execution are forever bound to that one version (modulo whole-transcript replay to switch to a different version, which might work with consensus but we can figure out the details later).